sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosItemWriteRetriesTest.java

```diff
@@ -0,0 +1,769 @@
+import com.azure.cosmos.models.PartitionKey;
+import com.azure.cosmos.rx.TestSuiteBase;
+import com.azure.cosmos.test.faultinjection.FaultInjectionCondition;
+import com.azure.cosmos.test.faultinjection.FaultInjectionConditionBuilder;
+import com.azure.cosmos.test.faultinjection.FaultInjectionConnectionType;
+import com.azure.cosmos.test.faultinjection.FaultInjectionOperationType;
+import com.azure.cosmos.test.faultinjection.FaultInjectionResultBuilders;
+import com.azure.cosmos.test.faultinjection.FaultInjectionRule;
+import com.azure.cosmos.test.faultinjection.FaultInjectionRuleBuilder;
+import com.azure.cosmos.test.faultinjection.FaultInjectionServerErrorResultBuilder;
+import com.azure.cosmos.test.faultinjection.FaultInjectionServerErrorType;
+import com.azure.cosmos.test.faultinjection.IFaultInjectionResult;
+import com.azure.cosmos.test.implementation.faultinjection.FaultInjectorProvider;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.testng.SkipException;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.UUID;
+import java.util.function.Supplier;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+public class CosmosItemWriteRetriesTest extends TestSuiteBase {
+    private static final ObjectMapper OBJECT_MAPPER = Utils.getSimpleObjectMapper();
+    private static final boolean WITH_INJECTION = true;
+    private static final boolean NO_INJECTION = false;
+    private static final boolean CONTENT_RESPONSE_ENABLED = true;
+    private static final boolean NO_CONTENT_RESPONSE = false;
+    private static final boolean WITH_PRECONDITION_CHECK = true;
+    private static final boolean NO_PRECONDITION_CHECK = false;
+    private static final Boolean DEFAULT_REQUEST_SUPPRESSION = null;
+    private static final Boolean ENFORCED_REQUEST_SUPPRESSION = true;
+    private static final Boolean NO_REQUEST_SUPPRESSION = false;
+    private static final WriteRetryPolicy NO_RETRIES = WriteRetryPolicy.DISABLED;
+    private static final WriteRetryPolicy RETRIES_WITHOUT_TRACKING_ID = new WriteRetryPolicy(true, false);
+    private static final WriteRetryPolicy RETRIES_WITH_TRACKING_ID = new WriteRetryPolicy(true, true);
+
+    private TracerUnderTest mockTracer;
+
+    @Factory(dataProvider = "clientBuildersWithDirectTcpSession")
+    public CosmosItemWriteRetriesTest(CosmosClientBuilder clientBuilder) {
+        super(clientBuilderWithReducedNetworkRequestTimeout(clientBuilder));
+    }
+
+    private static CosmosClientBuilder clientBuilderWithReducedNetworkRequestTimeout(
+        CosmosClientBuilder clientBuilder) {
+
+        clientBuilder.isContentResponseOnWriteEnabled();
```

Comment:
is the purpose to set the flag? ` clientBuilder.contentResponseOnWriteEnabled(true)`

Reply:
No -intend was to trigger intelli-sense :-)

Reply:
Fixed it in next iteration